src/Symfony/Component/Encryption/EncryptionInterface.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption;
+
+use Symfony\Component\Encryption\Exception\DecryptionException;
+use Symfony\Component\Encryption\Exception\EncryptionException;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+interface EncryptionInterface
+{
+    /**
+     * Generates new a Key to be used with encryption.
+     *
+     * Don't lose your private key and make sure to keep it a secret.
```

Comment:
i would write first to make it secret (and where/how to "store" it)
then to not loosing it :) 

also and related to the secret component that has a key rotate feature, will this kind of api be present here?

Reply:
Yes, you can use this API to rotate the secrets. It is basically just to decrypt everything, generate a new key, encrypt everything, save the new key and throw away the old one.